src/main/java/Jarvis/Parser.java

```diff
@@ -0,0 +1,215 @@
+
+    /**
+     * Constructs a new Parser object.
+     */
+    public Parser() {
+        validCommands = new ArrayList<>();
+        addValidCommands();
+    }
+
+    /**
+     * Adds the list of valid commands into the validCommands ArrayList.
+     */
+    private void addValidCommands() {
+        validCommands.add("list");
+        validCommands.add("bye");
+        validCommands.add("mark");
+        validCommands.add("uncheck");
+        validCommands.add("todo");
+        validCommands.add("deadline");
+        validCommands.add("event");
+        validCommands.add("find");
+    }
+
+    /**
+     * checks if the user inputted a valid command in the validCommand ArrayList.
+     *
+     * @param inputCommand commands that the user inputs.
+     * @return the valid command that the user inputted, an empty string otherwise.
+     */
+    // checks if command is valid and throws Jarvis.IncorrectJarvisCommandException
+    private static String isValidCommand(String inputCommand) {
+        // check if command is one of the valid keywords
+        boolean isValidCommand = false;
+        String validInputCommand = "";
+
+        for (String validCommand: validCommands) {
+            if (inputCommand.contains(validCommand)) {
+                isValidCommand = true;
+                validInputCommand = validCommand;
+                break;
+            }
+        }
+
+        if (!isValidCommand) {
+            try {
+                throw new IncorrectJarvisCommandException("Apologies Sir, the command " + inputCommand + " is not a valid command.");
+            } catch (IncorrectJarvisCommandException e) {
+                Ui.printListOfCommands(validCommands, inputCommand, e);
+            }
+        }
+        return validInputCommand;
+    }
+
+    /**
+     * checks if the command the user inputted has a wrong format, if the format of the command is wrong,
+     * print the details to the screen for the user to know how to correct it. This method assumes that
+     * isValidCommand has been run before this and the user has used a valid command. It just checks if
+     * the formatting of the command is correct.
+     *
+     * @param inputCommand the command the user inputted
+     * @param validInputCommand the valid command which the user inputted
+     */
+    // identifies which command has wrong formatting and prints feedback to user
+    public static void isWrongFormat(String inputCommand, String validInputCommand) {
+        if (validInputCommand.equals("")) { // if input is empty
+            return;
+        }
+
+        // since command is valid, check if formatting of the command is correct
+        boolean markMatch = inputCommand.matches("mark \\d+");
+        boolean uncheckMatch = inputCommand.matches("unmark \\d+");
+        boolean todoMatch = inputCommand.matches("todo .+");
+        boolean deadlineMatch = inputCommand.matches("deadline .+ /.+");
+        boolean eventMatch = inputCommand.matches("event .+ /.+ /.+");
```

Comment:
if possible try to use a prefix like is, has or was to make it clearer that it is a boolean